src/sentry/digests/utils.py

```diff
@@ -97,10 +99,20 @@ def get_event_from_groups_in_digest(digest: Digest) -> Iterable[Event]:
     }
 
 
-def build_custom_digest(original_digest: Digest, events: Iterable[Event]) -> Digest:
+def build_custom_digest(
+    original_digest: Digest, events: Iterable[Event], participant: RpcActor
+) -> Digest:
     """Given a digest and a set of events, filter the digest to only records that include the events."""
     user_digest: Digest = {}
     for rule, rule_groups in original_digest.items():
+        skip = False
+        rule_snoozes = RuleSnooze.objects.filter(Q(rule=rule))
+        for snooze in rule_snoozes:
+            if snooze.user_id is None or snooze.user_id == participant.id:
```

Comment:
Might actually make sense to include this clause in your query criteria. Then if there are any `rule_snooze` rows you can just continue.